### PR TITLE
[CP] Align the v26.7 release branch with the published images: release Dockerfiles, JAX alignment, and v26.7 docs (#1140, #1142, #1143, #1171, #1172)

### PR DESCRIPTION
**Pull request by amd-fuyuajin**
## Summary

Cherry-picks the five v26.7 release PRs onto `release/v26.7`, now that the branch point
has been advanced to `e7968675`.

| PR | What |
| -- | ---- |
| #1140 | mamba-370m config for MI325X |
| #1143 | `Dockerfile.jax-v26.7` |
| #1142 | `Dockerfile.primus-v26.7` — pins `PRIMUS_BRANCH=2631e68d`, so the image's build provenance is recorded on the branch even though the branch point has moved past it |
| #1172 | JAX alignment: `third_party/maxtext` → `b3c53763` (the commit the published MaxText image contains), `.gitmodules` tracking `release/v26.7`, the MaxDiffusion install hook, MI300X mixtral_8x22B batch size |
| #1171 | v26.7 documentation for both image families |

Grouped into one cherry-pick PR following the convention already used on
`release/v26.6`, where [#1160](https://github.com/AMD-AGI/Primus/pull/1160) landed #1052,
#1114 and #1120 together as a single `[CP]` commit.

## Testing

- **WAN 2.1 1.3B trains end to end on this branch** (`primus-cli container -- train
  pretrain --config examples/maxdiffusion/configs/MI300X/wan2.1_1.3b-pretrain.yaml`):
  20 steps, 357.698 TFLOP/s/device, loss 1.540, clean shutdown.
- `install_parity.py --check`: the bare-metal install scripts from #1171 match the v26.7
  Dockerfiles from #1142/#1143. This check is only meaningful because both land together —
  a reason not to split this PR.
- `check_links.py`: 0 new problems.
- `pre-commit run --all-files` and `tools/ci/check_version_consistency.py`: clean.
- The branch reports `## v26.7 (current)` and `__version__ = "26.7"`, where `release/v26.7`
  previously said v26.5.

### CODE REVIEW

**Review by GeneDer — APPROVED** [review]
We should have a PR for each cherry-picks, but just went to get this out ASAP, I will let it slip this time
